gtsam/geometry/tests/testCal3Bundler.cpp

```diff
@@ -89,6 +89,33 @@ TEST(Cal3Bundler, DcalibrateDefault) {
   CHECK(assert_equal(numerical2, Dp, 1e-5));
 }
 
+/* ************************************************************************* */
+TEST(Cal3Bundler, DuncalibratePrincipalPoint) {
+  Cal3Bundler K(5, 0, 0, 2, 2);
+  Matrix Dcal, Dp;
+  Point2 actual = K.uncalibrate(p, Dcal, Dp);
+  Point2 expected(12, 17);
+  CHECK(assert_equal(expected, actual, 1e-7));
+  Matrix numerical1 = numericalDerivative21(uncalibrate_, K, p);
+  Matrix numerical2 = numericalDerivative22(uncalibrate_, K, p);
+  CHECK(assert_equal(numerical1, Dcal, 1e-7));
+  CHECK(assert_equal(numerical2, Dp, 1e-7));
+}
+
+/* ************************************************************************* */
+TEST(Cal3Bundler, DcalibratePrincipalPoint) {
+  Cal3Bundler K(2, 0, 0, 2, 2);
```

Comment:
There’s also a global static variable by name K, just FYI in case you missed that. I don’t think it should cause any issues though.

Reply:
Yeah I know about that. The local scope will take precedence so it's all good.